storage/isolation_manager.rs

```diff
@@ -212,17 +137,75 @@ impl IsolationManager {
     }
 }
 
+fn resolve_concurrent(
```

Comment:
Previously `validate_concurrent`.

Not entirely happy with this name: this procedure does a lot of things, but the main ones are conflict detection, updating the state of puts, and potentially waiting for the predecessor to finish validation unless the commits are independent.